lib/mixins/property-effects.js

```diff
@@ -415,25 +421,43 @@ function runComputedEffects(inst, changedProps, oldProps, hasPaths) {
       Object.assign(/** @type {!Object} */ (changedProps), inst.__dataPending);
       inputProps = inst.__dataPending;
       inst.__dataPending = null;
+      // Ensure all computed properties are de-duped against the same turn
+      dedupeId--;
     }
   }
 }
 
+const TRANSITIVE_DEPENDENCY = '~transitive~dependency~';
+
 /**
  * Implements the "computed property" effect by running the method with the
  * values of the arguments specified in the `info` object and setting the
  * return value to the computed property specified.
  *
  * @param {!Polymer_PropertyEffects} inst The instance the effect will be run on
  * @param {string} property Name of property
- * @param {?Object} props Bag of current property changes
+ * @param {?Object} changedProps Bag of current property changes
  * @param {?Object} oldProps Bag of previous values for changed properties
  * @param {?} info Effect metadata
  * @return {void}
  * @private
  */
-function runComputedEffect(inst, property, props, oldProps, info) {
-  let result = runMethodEffect(inst, property, props, oldProps, info);
+function runComputedEffect(inst, property, changedProps, oldProps, info) {
+  if (orderedComputed) {
+    // Compute any computed dependencies first; this recurses through the
+    // dependency graph to ensure computed properties are never computed with
+    // dependencies that may become invalidated later in this turn
+    computeDependencies(inst, info, changedProps, oldProps);
+    // If the dependency was not transitive, it's definitely dirty and needs to
+    // be computed; if it is transitive, check its arguments against the current
+    // changed props and only re-compute if it is dirty
+    if (property === TRANSITIVE_DEPENDENCY && 
+        !computedPropertyIsDirty(inst, info, changedProps)) {
```

Comment:
Did you consider instead capturing the argument values before computing dependencies and comparing those explicitly?

Reply:
Resolving as outdated.